src/utils/styles.js

```diff
@@ -8,4 +12,71 @@ module.exports = {
     return AutoPrefix.all(mergedStyles);
   },
 
+  // This function ensures that `style` supports both ltr and rtl directions by checking 
+  //   `styleConstants` in `muiTheme` and replacing attribute keys if necessary.
+  ensureDirection(muiTheme, style) {
+    if (process.env.NODE_ENV !== 'production') {
+      if (style.didFlip) {
+        console.warn(new Error('You\'re calling `ensureDirection` on the same style object twice.'));
+      }
+      style = ImmutabilityHelper.merge({
+        didFlip: 'true',
+      }, style);
+    }
+
+    if (!muiTheme.isRtl) return style; // Performance hack!
```

Comment:
`Left to right is the default` doesn't imply that this line of code is there for optimization. Maybe:
`Skip further processing to improve performance, since ltr is the default.`